src/events/kafkajs.js

```diff
@@ -157,18 +156,19 @@ function kafkaConnectWrapper(originalConnectFunction) {
 }
 
 /**
- * Wraps the kafkajs producer creation
- * @param {Function} wrappedFunction The kafkajs producer function
+ * Wraps the kafkajs producer creation and wrapping send function.
+ * @param {Function} producerFunction The kafkajs producer function
  * @returns {Function} The wrapped function
  */
-function kafkaProducerWrapper(wrappedFunction) {
+function kafkaProducerWrapper(producerFunction) {
     return function internalKafkaProducerWrapper(options) {
-        // patching send request.
-        const producer = wrappedFunction.apply(this, [options]);
-        const patchedSend = messages => kafkaSendMiddleware(messages, producer);
-        producer.originalSend = producer.send;
-        producer.send = patchedSend.bind(producer);
-        return producer;
+        const producerResponse = producerFunction.apply(this, [options]);
+        try {
+            shimmer.wrap(producerResponse, 'send', () => wrapKafkaSendFunction(producerResponse.send));
```

Comment:
Can you add a protection so this shimmer will happen only once?

Reply:
added